### ISSUE TITLE
side effect of admin pwd chagne

### ISSUE THREAD

**Issue by drehpehs** [goharbor/harbor#14193]
Hi Sir, we host Harbor as our Registry with two instances connected to external database and redis server. And recently we try to change the admin password by Harbor UI. Everything is fine with new password login, but we found the replication and API call don't work well sometimes. 
We have another Harbor as backup Registry, we setup replication policy with admin account as access ID and secret. Since we change the admin, the replication failed sometimes and the error message is **401 UNAUTHORIZED**
Also we have Jenkins to do some Harbor API call and store admin pwd in credential data. After we change the password, the API call failed sometimes as well. 

We don't know the root cause of it. And it bothered us a lot. Could you help us to check it? Or any suggestion to resolve this problem? Thank you in advance.

Versions:
harbor version: [2.1.1]
docker engine version: [18.09.0]
docker-compose version: [1.24.1]

**Comment by bitsf** [comment]
can you login remote harbor from UI after change password ?

**Comment by drehpehs** [comment]
@bitsf 
Yes, I can. The UI is working well with new password.
It only happened to API call or replication sometimes.

**Comment by bitsf** [comment]
Do you mean have two harbor connected to same one db and redis?
And how did you change the password?

**Comment by drehpehs** [comment]
@bitsf 
Yes, we configured two harbor instances connected to one DB and redis.
We changed the password by Harbor UI. (admin account on the right top corner)

**Comment by drehpehs** [comment]
@bitsf 
Excuse me, any update?

**Comment by bitsf** [comment]
user password is just saving in db, and there's no cache.
maybe you can also check if the env `CORE_SECRET` `REGISTRY_CREDENTIAL_PASSWORD` in two harbor-core pods are same, and file `/etc/registry/passwd` in registry pods are same

**Comment by reasonerjt** [comment]
@drehpehs you mentioned 
```
the replication failed sometimes
```

So could you check if the API call fails when the request is routed to one of the instances?

**Comment by drehpehs** [comment]
@bitsf @reasonerjt 
Hi, sorry for the late reply.
Actually we found the problem, there was somewhere still using old admin pwd to call the API frequently.
`failed to authenticate admin: Failed to authenticate user, due to error 'Invalid credentials'`

However, the admin account was locked because of this (we checked it with debug mode). So we still got 401 UNAUTHORIZED even we used the correct(new) admin pwd.
`2021-03-31T08:41:24Z [DEBUG] [/core/auth/authenticator.go:146]: admin is locked due to login failure, login failed`

Afterwards, we removed/changed all the place calling API or remote registry configuration. This problem is resolved now.
Thanks for your concerning and reply.
